src/lib/operators/index_scan.hpp

```diff
@@ -21,13 +20,15 @@ class AbstractTask;
  */
 class IndexScan : public AbstractReadOnlyOperator {
  public:
-  IndexScan(const std::shared_ptr<const AbstractOperator>& input_operator, const ChunkIndexType index_type,
-            const std::vector<ColumnID>& left_column_ids, const PredicateCondition predicate_condition,
-            const std::vector<AllTypeVariant>& right_values, const std::vector<AllTypeVariant>& right_values2 = {});
+  IndexScan(const std::shared_ptr<const AbstractOperator>& input_operator, const ColumnID indexed_column_id,
+            const PredicateCondition predicate_condition, const AllTypeVariant scan_value);
 
   const std::string& name() const final;
 
-  // If set, only the specified chunks will be scanned. See TableScan::excluded_chunk_ids for usage.
+  // Must not be empty because only the specified chunks will be scanned. See TableScan::excluded_chunk_ids for usage.
+  // Note: These ChunkIDs are reffering to the ChunkIDs of a (!! non-pruned !!) data table to which the index belongs.
+  // They do not refer to the ChunkIDs output by potentially chunk-pruned GetTable operartors (i.e., the index scan's
+  // input).
```

Comment:
```
// Must not be empty because only the specified chunks will be scanned. See TableScan::excluded_chunk_ids for usage.
// Note: These ChunkIDs are refering to the ChunkIDs of the input operator (i.e., GetTable) at optimization-time.
// Due to dynamic pruning, the included ChunkIDs must potentially be updated for further pruned chunks.
```

Better?